### PR TITLE
expose the flag for skipping trust domain validation

### PR DESCRIPTION
**Pull request by yangminzhu**
Please provide a description for what this PR is for.

Fixes https://github.com/istio/istio/issues/15631

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/features/pilot.go [hunk 1]
```diff
@@ -157,6 +157,13 @@ var (
 		false,
 		"Use the Istio JWT filter for JWT token verification.")
 
+	// SkipValidateTrustDomain tells the server proxy to not to check the peer's trust domain when
+	// mTLS is enabled in authentication policy.
+	SkipValidateTrustDomain = env.RegisterBoolVar(
+		"SKIP_VALIDATE_TRUST_DOMAIN",
```
Comment (hzxuzhonghu) [inline pilot/pkg/features/pilot.go]: To keep consistency, suggest adding `PILOT_` prefix

Comment (yangminzhu) [reply pilot/pkg/features/pilot.go]: Done.

**Comment by rlenglet** [comment]
This needs to get backported into 1.3 ASAP. Can a Pilot maintainer from @istio/wg-networking-maintainers review? Thanks!

**Comment by rlenglet** [comment]
`istio-racetest-master` failed with:
```
*** Test killed with quit: ran too long (10m0s).
FAIL	istio.io/istio/galley/pkg/config/meshcfg	600.009s
```
`istio-unit-tests-master` failed with:
```
*** Test killed with quit: ran too long (10m0s).
FAIL	istio.io/istio/galley/pkg/config/meshcfg	600.052s
```
Those may be environment issues.

**Comment by yangminzhu** [comment]
I will rebase to see if it fixes the racetest and unit tests, it should be irrelevant with this PR.

**Pull request merged**